go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -339,13 +340,19 @@ func (c *Controller) syncServices(key string) error {
 				}
 				// Configure the NodePort in each Node Gateway Router
 				for _, gatewayRouter := range gatewayRouters {
-					lbID, err := gateway.GetGatewayLoadBalancer(gatewayRouter, svcPort.Protocol)
+					gatewayLB, err := gateway.GetGatewayLoadBalancer(gatewayRouter, svcPort.Protocol)
 					if err != nil {
 						klog.Warningf("Service Sync: Gateway router %s does not have load balancer (%v)",
 							gatewayRouter, err)
 						// TODO: why continue? should we error and requeue and retry?
 						continue
 					}
+					workerNode := strings.TrimPrefix(gatewayRouter, "GR_")
+					workerLB, err := loadbalancer.GetWorkerLoadBalancer(workerNode, svcPort.Protocol)
+					if err != nil {
+						klog.Errorf("Worker switch %s does not have load balancer (%v)", workerNode, err)
+						continue
```

Comment:
inside this loop if you return an error it enqueues the service and retries again, there is a rate limit and a max number of retries.
The question is should we ignore this workerif it doesn't have a worker LB or should we retry? if the later we should return an error here

Reply:
so in 5609c3a5eb0afba767839f279ceb80f5bd90d739 I moved all of this to its own function, however the behavior is the same. I feel like I intentionally did this for a reason, but I cannot remember what it was. Looking at the function "createPerNodeVIPs" from the later commit the call happens 3 times:

1. Shared GW - has host endpoints, need to configure worker LB and GR LB
2. Node Port - configure GR (and worker LB if shared gw)
3. External IPs - configure GR (and worker LB if shared gw)

So I think based on what I see it is justified to expect the node has the worker switch, and if not return an error. I'll update the code in the following commit/function to reflect that.